Server/Controller/UserController.js

```diff
@@ -0,0 +1,78 @@
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import { pool } from '../dbconfig';
+
+dotenv.config();
+//=================================================== Sing Up ===============================================
+const signUpUser = async(req, res) => {
+    
+  const take = req.body.Email;
+  const findemail = 'SELECT * FROM users WHERE email= $1';
+  const user = await pool.query(findemail, [take]);
+  if (user.rows[0]) {
+      return res.status(400).send({
+          status: 400,
+          message: 'Email exist',
+      });
+  }
+  const hide = bcrypt.hashSync(req.body.password.trim(), 10);
```

Comment:
if you trim the password, the password might not match what a user would input.

Only use trim to ensure the user just doesn't enter empty spaces alone. Empty spaces with other letters can be a valid password